Mathlib/Data/Finsupp/Basic.lean

```diff
@@ -872,33 +872,27 @@ section Filter
 
 section Zero
 
-variable [Zero M] (p : α → Prop) (f : α →₀ M)
+variable [Zero M] (p : α → Prop) [DecidablePred p] (f : α →₀ M)
 
 /--
 `Finsupp.filter p f` is the finitely supported function that is `f a` if `p a` is true and `0`
 otherwise. -/
-def filter (p : α → Prop) (f : α →₀ M) : α →₀ M
-    where
-  toFun a :=
-    haveI := Classical.decPred p
-    if p a then f a else 0
-  support :=
-    haveI := Classical.decPred p
-    f.support.filter fun a => p a
+def filter (p : α → Prop) [DecidablePred p] (f : α →₀ M) : α →₀ M where
```

Comment:
I understand that if somebody wants to use `Finsupp.filter`, either their predicate is decidable (possibly by adding that to the instances of the theorem), and then everything's OK, otherwise one invokes `Classical.decPred` before using it, as you do below in `coeff_weightedHomogeneousComponent `.
It would probably be interesting that  there be a general page on mathlib's documentation about this, hoping that everybody will care and agree.

Reply:
Unfortunately I don't think we have any good mechanism at the moment to host such a general documentation page; library notes are not yet shown in doc-gen...

Reply:
Would the mathlib overview host a section on decidability?

Reply:
> Unfortunately I don't think we have any good mechanism at the moment to host such a general documentation page; library notes are not yet shown in doc-gen...

At least temporarily the github wiki feature could host things like this?